middleware/security/jwt/jwt.go

```diff
@@ -91,18 +212,41 @@ func New(validationKeys interface{}, validationFunc goa.Middleware, scheme *goa.
 
 			incomingToken := strings.Split(val, " ")[1]
 
+			var jwtKeyName string
+			if resolver.jwtKeyNameField != "" {
+				jwtKeyName = req.Header.Get(resolver.jwtKeyNameField)
+			}
+
+			// Make a copy of the current keys in the KeyResolver's key map
+			resolver.Lock()
```

Comment:
It would probably make sense to use a `RWLock` instead of just a lock since here we only need a read lock. This seems like a valuable change because this code runs for each request and using a simple lock forces synchronization which could impact performance and is not needed.